### PR TITLE
add sequenceS (fixes #2)

### LINKED ISSUE #2: Implement sequenceS
This is the follow up of https://github.com/gcanti/fp-ts/issues/688

### PR DESCRIPTION
**Pull request by sledorze**
@gcanti ready to be discussed further

### CODE REVIEW

File: src/sequence-s.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  F: Apply2C<F, L>
+): <R extends Record<string, Type2<F, L, any>>>(
+  r: EnforceNonEmptyRecord<R>
+) => Type2<F, L, { [K in keyof R]: R[K] extends Type2<F, any, infer A> ? A : never }>
+export function sequenceS<F extends URIS>(
+  F: Apply1<F>
+): <R extends Record<string, Type<F, any>>>(
+  r: EnforceNonEmptyRecord<R>
+) => Type<F, { [K in keyof R]: R[K] extends Type<F, infer A> ? A : never }>
+export function sequenceS<F extends URIS>(
```
Comment (gcanti) [inline src/sequence-s.ts]: should be

```diff
+export function sequenceS<F>(
-export function sequenceS<F extends URIS>(
  F: Apply<F>
): (r: Record<string, HKT<F, any>>) => HKT<F, Record<string, any>>
```

and should also be added as overloading

Comment (sledorze) [reply src/sequence-s.ts]: Oh yes!

**Comment by gcanti** [comment]
@sledorze for what concerns the filename, we could use the combinator name (`sequenceS.ts`)

**Comment by gcanti** [comment]
Thanks @sledorze, did you tested it against your code base? (you already had a version of this IIRC)

**Comment by sledorze** [comment]
@gcanti will try this week (a bit on hurry to deliver ATM)

**Pull request merged**